tm2/pkg/crypto/keys/client/export.go

```diff
@@ -88,25 +89,25 @@ func execExport(cfg *ExportCfg, io commands.IO) error {
 		)
 	}
 
-	var (
-		armor     string
-		exportErr error
-	)
+	var keyArmor string
 
 	if cfg.Unsafe {
-		// Generate the unencrypted armor
-		armor, exportErr = kb.ExportPrivKeyUnsafe(
-			cfg.NameOrBech32,
-			decryptPassword,
-		)
-
-		privk, err := kb.ExportPrivateKeyObject(cfg.NameOrBech32, decryptPassword)
+		privk, err := kb.ExportPrivKey(cfg.NameOrBech32, decryptPassword)
 		if err != nil {
-			panic(err)
+			return fmt.Errorf("unable to export private key, %w", err)
 		}
 
 		io.Printf("privk:\n%x\n", privk.Bytes())
+
+		// Generate the private key armor
+		keyArmor = armor.ArmorPrivateKey(privk)
 	} else {
+		// Generate the encrypted armor
+		privk, err := kb.ExportPrivKey(cfg.NameOrBech32, decryptPassword)
+		if err != nil {
+			return fmt.Errorf("unable to export private key, %w", err)
+		}
```

Comment:
This block is duplicated in the if (95-98) and else (106-109) branches. You could place it before the if to deduplicate it.

Reply:
Good catch 💯 

[3b571b5](https://github.com/gnolang/gno/pull/3285/commits/3b571b5927e6df79e853bea6d0577405ed826521)